components/view_image/view_image.tsx

```diff
@@ -247,28 +286,34 @@ export default class ViewImageModal extends React.PureComponent {
     }
 
     handleModalClose = () => {
-        this.props.onModalDismissed();
-        this.setState({scale: Utils.fillArray(ZoomSettings.DEFAULT_SCALE, this.props.fileInfos.length)});
+        if (this.props?.onModalDismissed) {
+            this.props.onModalDismissed();
+        }
+
+        this.setState({scale: Utils.fillArray(ZoomSettings.DEFAULT_SCALE, this.props?.fileInfos?.length)});
     }
 
-    render() {
-        if (this.props.fileInfos.length < 1 || this.props.fileInfos.length - 1 < this.state.imageIndex) {
+    public render(): JSX.Element | null {
+        const fileInfosLength = this.props.fileInfos?.length ?? 0;
+        const imageIndex = this.state?.imageIndex ?? 0;
```

Comment:
if you ensure `this.state.imageIndex` can't be anything but a number you don't need this.